shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/repositories/OnboardingRepository.kt

```diff
@@ -0,0 +1,60 @@
+package com.mbta.tid.mbta_app.repositories
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringSetPreferencesKey
+import co.touchlab.skie.configuration.annotations.DefaultArgumentInterop
+import com.mbta.tid.mbta_app.model.OnboardingScreen
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+interface IOnboardingRepository {
+    suspend fun getPendingOnboarding(): List<OnboardingScreen>
+
+    suspend fun markOnboardingCompleted(screen: OnboardingScreen)
+}
+
+class OnboardingRepository : IOnboardingRepository, KoinComponent {
+    private val accessibilityStatus: IAccessibilityStatusRepository by inject()
+    private val dataStore: DataStore<Preferences> by inject()
+
+    private val onboardingCompletedKey = stringSetPreferencesKey("onboardingCompleted")
```

Comment:
suggestion(non-blocking): "onboardingCompleted" as a string name reads to me as a boolean value, event though I know it isn't. Consider calling it something like `onboardingCompletedScreens` for clarity